force-app/main/default/classes/SOQL.cls

```diff
@@ -39,20 +41,28 @@ public with sharing class SOQL {
      * @return A list of Contact records.
      * Update the corresponding test class and testGetContactsByFirstName method to cover this method
      */
-    //[public | private ] [static] data_type method_name (input parameters) {
-    //  The body of the method
-    //}
-
-    /**
+    public static List<Contact> getContactsByFirstName() {
+        List<Contact> contacts = new List<Contact>([SELECT Id, Name, FirstName, LastName 
+                                                    FROM Contact WHERE FirstName = 'John']);
+        return contacts;
+    }
+    
+        /**
      * Question 3
      * Update/Uncomment the template code to create a method
      * Returns the Contact Id for all contacts with last name of Smith or Jones
      * @return A list of Contact Ids.
      * Update the corresponding test class and testGetContactIdsByLastName method to cover this method
      */
-    //[public | private ] [static] data_type method_name (input parameters) {
-    //  The body of the method
-    //}
+    public static List<Id> getContactIdsByLastName() {
+        List<Id> contactIds = new List<Id>();
+        List<Contact> contacts = new List<Contact>([SELECT Id, LastName FROM Contact 
+                                                    WHERE LastName = 'Smith' OR LastName = 'Jones']);
+        for (Contact contact : contacts) {
```

Comment:
Great job! You can also loop over the query results directly like this:
`for (Contact contact : [SELECT Id, LastName FROM Contact WHERE LastName = 'Smith' OR LastName = 'Jones']) {`